modules/mono/editor/GodotTools/GodotTools/GodotSharpEditor.cs

```diff
@@ -455,7 +455,7 @@ public override void _EnablePlugin()
             _menuPopup.IdPressed += _MenuOptionPressed;
 
             // External editor settings
-            EditorDef("mono/editor/external_editor", ExternalEditorId.None);
+            EditorDef("mono/editor/external_editor", (int)ExternalEditorId.None);
```

Comment:
I'm aware. TBH, I'm not entirely happy about it. This wouldn't be a problem if casting an enum to a smaller type would result in an error, but it doesn't. Unfortunately, operator overloads can't be generic, so we can't have one for enums to Variant.

I was thinking of adding a helper method. Either an extension method (`ExternalEditorId.None.ToVariant()`), or a normal one (`EnumUtils.EnumValue(ExternalEditorId.None)`). For the latter, we would rely on `global using static EnumUtils;` to shorten it: `EnumValue(ExternalEditorId.None)`. None of these would be better than implicit conversion, but at least they're not error-prone.

Reply:
There's an issue about using enums as Variants (https://github.com/godotengine/godot/issues/66013) but I'm not sure using a static class would make things better.

Also, wouldn't using `Variant.From(ExternalEditorId.None)` work the same way? I don't see the benefit of an enum specific API if `Variant.From` works, specially since what people really want is to not need to call anything at all (implicit conversion) but that's not possible.

This is similar to `Callable` and implicit conversion from delegates or method groups. It's not possible so we just have to live with `Callable.From`, maybe it's not so bad to require using `Variant.From`. Also, the fact that both APIs are named the same probably helps.

Reply:
Ah, right. We already have `Variant.From`. Perhaps a short alternative for this would be a good idea, thanks to `global using static`, e.g.: `Var` for `Variant.From`, and `Bind` for `Callable.From`. Or maybe it would just cause confusion.

Reply:
Ah, right. We already have `Variant.From`. Perhaps it would be a good idea to have a short alternative for this, by benefiting from `global using static`, e.g.: `Var` for `Variant.From`, and `Bind` for `Callable.From`. Or maybe it would just cause confusion.

Reply:
Honestly, I think `Variant.From` and `Callable.From` are short enough, and I'm not a big fan of having too many ways of doing the same thing, I think it would indeed cause confusion.

Reply:
Changed the cast to `Variant.From`.